### ISSUE TITLE
Incorrect handling of main generates broken code

### ISSUE THREAD

**Issue by andreaTP** [scala-js/scala-js#3082]
A minimal repro is:
Scala.Js 0.6.19
build.sbt:
```scala
lazy val root = project
  .in(file("."))
  .settings(
    scalaVersion := "2.12.2",
    scalaJSUseMainModuleInitializer in Global := true
  )
  .enablePlugins(ScalaJSPlugin)
```
Main.scala:
```scala
package ui

object Main {
  def main(args: Array[String]): Unit = {
    println("Hello world")
  }
}
```
running `sbt run` I got:
```
[info] Running ui.Main
[error] [stdin]:40
[error]          global["Object"] === Object) ? global : this)["ui"]["Main"]().main();
[error]                                                             ^
[error] 
[error] TypeError: Cannot read property 'Main' of undefined
[error]     at [stdin]:40:61
[error]     at ContextifyScript.Script.runInThisContext (vm.js:23:33)
[error]     at Object.runInThisContext (vm.js:95:38)
[error]     at Object.<anonymous> ([stdin]-wrapper:6:22)
[error]     at Module._compile (module.js:571:32)
[error]     at evalScript (bootstrap_node.js:391:27)
[error]     at Socket.<anonymous> (bootstrap_node.js:188:13)
[error]     at emitNone (events.js:91:20)
[error]     at Socket.emit (events.js:188:7)
[error]     at endReadableNT (_stream_readable.js:975:12)
org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: Node.js exited with code 1
	at org.scalajs.jsenv.ExternalJSEnv$AbstractExtRunner.waitForVM(ExternalJSEnv.scala:119)
	at org.scalajs.jsenv.ExternalJSEnv$ExtRunner.run(ExternalJSEnv.scala:168)
	at org.scalajs.sbtplugin.ScalaJSPluginInternal$.org$scalajs$sbtplugin$ScalaJSPluginInternal$$jsRun(ScalaJSPluginInternal.scala:718)
	at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$73$$anonfun$apply$50$$anonfun$apply$51.apply(ScalaJSPluginInternal.scala:860)
	at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$73$$anonfun$apply$50$$anonfun$apply$51.apply(ScalaJSPluginInternal.scala:854)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[trace] Stack trace suppressed: run last compile:run for the full output.
[error] (compile:run) org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: Node.js exited with code 1
```

**Comment by sjrd** [comment]
This part is wrong:
```scala
scalaJSUseMainModuleInitializer in Global := true
```
The `in Global` should not be there, since this setting is only relevant per-project.

Arguably, we should give a better error message than that. The thing is that the launcher generation does not check whether a discovered main object is `js.JSApp` style. It will happily generate a launcher (persisted or not) for a new-style main object, but the generated launcher will be broken. It should instead check the style of main object, and fail if it is new style.

**Comment by andreaTP** [comment]
I cannot agree more, thanks! 👍

**Comment by sjrd** [comment]
Fixed in 39567a7bfc7184bccfcc8a849334a180faee16d2.
